src/main/java/emissary/core/BaseDataObject.java

```diff
@@ -37,7 +42,7 @@
     /* Including this here make serialization of this object faster. */
     private static final long serialVersionUID = 7362181964652092657L;
 
-    /* Our payload */
+    /* Actual data - migrate away from this towards byte channels. */
```

Comment:
This gives the impression that the plan is to eventually get to the exclusive use of ByteChannel rather than having an option to use either a ByteChannel or a standard byte[]. Is that a correct assumption?

Reply:
That is my understanding.

Reply:
@jdcove2 Should we be concerned about performance degredation when we are using channels for all the small stuff that doesn't really need them?  This may be an argument for the configurable option that @mjwall mentioned earlier... then for certain copilot pools that we know will only have good data we can configure just use arrays and for the rest configure to use these.

Reply:
1. The performance of pulling data directly from a byte[] and pulling data from an SBC that wraps a byte[] should be similar.
2. If there is a specific usage pattern (e.g. scanning linearly from the first byte to the last byte) then a "cache" can usually be added to an SBC to get similar performance to a byte[].
3. If the intent is to keep both byte[] and SBC, then Emissary/Burrito/Other will end up being a mixture of implementations that use byte[], SBC, or both, where the "byte[]" implementations will force the data to be put on the heap and limited to 2^31 in size and the "both" implementations will have "if" statements around multiple implementations.
4. I don't see a need for a "switch" as the setData(byte[])/getData(byte[]) BDO functionality in this PR is the same as the current functionality.